### DISCUSSION TITLE
Erreur MAJ DB 1.9.7 => 0400003

### DISCUSSION THREAD

**Discussion by Fred12400** [JeromeDevome/GRR#312] [Q&A] [answered]
Bonjour,

Lors de la mise à jour de mon GRR 1.9.7 vers 4.0.3, le script de MAJ de la base de données a rencontré des erreurs, est ce bloquant ?

Je voulais poser ma question sur le site https://site.devome.com/fr/kunena/index mais je n'ai pas réussi à m'inscrire, le Captchat tourne indéfiniment sans possibilité de le valider.

Merci

Mise à jour de la base de données
ATTENTION : Votre base de données ne semble pas être à jour. Numéro de version de la base de données : 1.9.7
Cliquez sur le bouton suivant pour effectuer la mise à jour vers la version 0400003
Résultat de la mise à jour < GRR 4
1.9.7 => 3.5.1
Mise à jour jusqu'à la version 2.0.0 :
Ok !
Mise à jour jusqu'à la version 3.0.0 :
Ok !
Mise à jour jusqu'à la version 3.1.0 :
Ok !
Mise à jour jusqu'à la version 3.2.0 :
Ok !
Mise à jour jusqu'à la version 3.3.0 :
Ok !
Mise à jour jusqu'à la version 3.3.1 :
Ok !
Mise à jour jusqu'à la version 3.4.0 :
Ok !
Mise à jour jusqu'à la version 3.4.1 :
Ok !
Mise à jour jusqu'à la version 3.4.2 :
Ok !
Mise à jour jusqu'à la version 3.4.3 RC0:
Erreur sur la requête : ALTER TABLE grr_log CHANGE `REMOTE_ADDR` `REMOTE_ADDR` VARCHAR(40) NOT NULL DEFAULT '' (1067 : Invalid default value for 'START')
Mise à jour jusqu'à la version 3.5.0 RC0:

Passage de la base en utf8mb4 COLLATE utf8mb4_unicode_ci
character_set_database est réglé à latin1
Passage de character_set_database à  utf8mb4 COLLATE utf8mb4_unicode_ci
Ok !
Passage des tables en utf8mb4 COLLATE utf8mb4_unicode_ci
Passage de grr_area en utf8mb4 COLLATE utf8mb4_unicode_ci en cours. Ok !
Passage de grr_area_periodes en utf8mb4 COLLATE utf8mb4_unicode_ci en cours. Ok !
Passage de grr_calendar en utf8mb4 COLLATE utf8mb4_unicode_ci en cours. Ok !
Passage de grr_calendrier_feries en utf8mb4 COLLATE utf8mb4_unicode_ci en cours. Ok !
Passage de grr_calendrier_jours_cycle en utf8mb4 COLLATE utf8mb4_unicode_ci en cours. Ok !
Passage de grr_calendrier_vacances en utf8mb4 COLLATE utf8mb4_unicode_ci en cours. Ok !
Passage de grr_correspondance_statut en utf8mb4 COLLATE utf8mb4_unicode_ci en cours. Ok !
Passage de grr_entry en utf8mb4 COLLATE utf8mb4_unicode_ci en cours. Ok !
Passage de grr_entry_moderate en utf8mb4 COLLATE utf8mb4_unicode_ci en cours. Erreur sur la requête : ALTER TABLE grr_entry_moderate CONVERT TO CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci (1292 : Incorrect datetime value: '0000-00-00 00:00:00' for column 'timestamp' at row 636)
Erreur sur la requête : ALTER TABLE grr_entry_moderate CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci (1292 : Incorrect datetime value: '0000-00-00 00:00:00' for column 'timestamp' at row 636)
Passage de grr_j_mailuser_room en utf8mb4 COLLATE utf8mb4_unicode_ci en cours. Ok !
Passage de grr_j_site_area en utf8mb4 COLLATE utf8mb4_unicode_ci en cours. Ok !
Passage de grr_j_type_area en utf8mb4 COLLATE utf8mb4_unicode_ci en cours. Ok !
Passage de grr_j_user_area en utf8mb4 COLLATE utf8mb4_unicode_ci en cours. Ok !
Passage de grr_j_user_room en utf8mb4 COLLATE utf8mb4_unicode_ci en cours. Ok !
Passage de grr_j_useradmin_area en utf8mb4 COLLATE utf8mb4_unicode_ci en cours. Ok !
Passage de grr_j_useradmin_site en utf8mb4 COLLATE utf8mb4_unicode_ci en cours. Ok !
Passage de grr_j_userbook_room en utf8mb4 COLLATE utf8mb4_unicode_ci en cours. Ok !
Passage de grr_log en utf8mb4 COLLATE utf8mb4_unicode_ci en cours. Erreur sur la requête : ALTER TABLE grr_log CONVERT TO CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci (1067 : Invalid default value for 'START')
Erreur sur la requête : ALTER TABLE grr_log CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci (1067 : Invalid default value for 'START')
Passage de grr_modulesext en utf8mb4 COLLATE utf8mb4_unicode_ci en cours. Ok !
Passage de grr_overload en utf8mb4 COLLATE utf8mb4_unicode_ci en cours. Ok !
Passage de grr_page en utf8mb4 COLLATE utf8mb4_unicode_ci en cours. Ok !
Passage de grr_participants en utf8mb4 COLLATE utf8mb4_unicode_ci en cours. Ok !
Passage de grr_repeat en utf8mb4 COLLATE utf8mb4_unicode_ci en cours. Ok !
Passage de grr_room en utf8mb4 COLLATE utf8mb4_unicode_ci en cours. Ok !
Passage de grr_setting en utf8mb4 COLLATE utf8mb4_unicode_ci en cours. Ok !
Passage de grr_site en utf8mb4 COLLATE utf8mb4_unicode_ci en cours. Ok !
Passage de grr_type_area en utf8mb4 COLLATE utf8mb4_unicode_ci en cours. Ok !
Passage de grr_utilisateurs en utf8mb4 COLLATE utf8mb4_unicode_ci en cours. Ok !
Mise à jour jusqu'à la version 3.5.1:
Erreur sur la requête : ALTER TABLE grr_log CHANGE LOGIN LOGIN varchar(190) CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci NOT NULL default '' (1067 : Invalid default value for 'START')
Erreur sur la requête : ALTER TABLE grr_entry_moderate CHANGE login_moderateur login_moderateur varchar(190) CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci NOT NULL default '' ; (1292 : Incorrect datetime value: '0000-00-00 00:00:00' for column 'timestamp' at row 636)
Erreur sur la requête : ALTER TABLE grr_entry_moderate CHANGE create_by create_by varchar(190) CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci NOT NULL default '' ; (1292 : Incorrect datetime value: '0000-00-00 00:00:00' for column 'timestamp' at row 636)
Erreur sur la requête : ALTER TABLE grr_entry_moderate CHANGE beneficiaire beneficiaire varchar(190) CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci NOT NULL default '' ; (1292 : Incorrect datetime value: '0000-00-00 00:00:00' for column 'timestamp' at row 636)
Résultat de la mise à jour GRR 4 et +
4000 => 0400003
Mise à jour jusqu'à la version 4.0.0 :
Ok !
Mise à jour jusqu'à la version 4.1.0 :
Ok !
Mise à jour jusqu'à la version 4.2.0 :
Ok !

**Comment by ynaessens** [comment, marked as answer]
Bonjour,
effectivement, je pense que ça ne va pas bien marcher ainsi.
J'ai apporté un correctif à la fonction de mise à jour, disponible à ce commit : https://github.com/JeromeDevome/GRR/commit/482f3a153acd6865d31b54c478581a0153658ba0
Pouvez-vous confirmer que cela résout votre problème ?
Cordialement,
YN

**Comment by Fred12400** [comment]
Bonjour,

Merci pour la mise à disposition rapide de ce patch.

Il me reste des erreurs sur certaines entrées de la table grr_entry_moderate, je suppose que cela ne doit pas être bloquant.
Une dernière erreur à la fin sur la création de la table grr_j_userbook_room qui existe déjà.

Résultat de la mise à jour < GRR 4
1.9.7 => 3.5.1
Mise à jour jusqu'à la version 2.0.0 :
Ok !
Mise à jour jusqu'à la version 3.0.0 :
Ok !
Mise à jour jusqu'à la version 3.1.0 :
Ok !
Mise à jour jusqu'à la version 3.2.0 :
Ok !
Mise à jour jusqu'à la version 3.3.0 :
Ok !
Mise à jour jusqu'à la version 3.3.1 :
Ok !
Mise à jour jusqu'à la version 3.4.0 :
Ok !
Mise à jour jusqu'à la version 3.4.1 :
Ok !
Mise à jour jusqu'à la version 3.4.2 :
Ok !
Mise à jour de la table grr_log:
Ok !
Mise à jour de la table grr_entry_moderate:
Ok !
Mise à jour jusqu'à la version 3.4.3 RC0:
Ok !
Mise à jour jusqu'à la version 3.5.0 RC0:

Passage de la base en utf8mb4 COLLATE utf8mb4_unicode_ci
character_set_database est réglé à latin1
Passage de character_set_database à  utf8mb4 COLLATE utf8mb4_unicode_ci
Ok !
Passage des tables en utf8mb4 COLLATE utf8mb4_unicode_ci
Passage de grr_area en utf8mb4 COLLATE utf8mb4_unicode_ci en cours. Ok !
Passage de grr_area_periodes en utf8mb4 COLLATE utf8mb4_unicode_ci en cours. Ok !
Passage de grr_calendar en utf8mb4 COLLATE utf8mb4_unicode_ci en cours. Ok !
Passage de grr_calendrier_feries en utf8mb4 COLLATE utf8mb4_unicode_ci en cours. Ok !
Passage de grr_calendrier_jours_cycle en utf8mb4 COLLATE utf8mb4_unicode_ci en cours. Ok !
Passage de grr_calendrier_vacances en utf8mb4 COLLATE utf8mb4_unicode_ci en cours. Ok !
Passage de grr_correspondance_statut en utf8mb4 COLLATE utf8mb4_unicode_ci en cours. Ok !
Passage de grr_entry en utf8mb4 COLLATE utf8mb4_unicode_ci en cours. Ok !
**Passage de grr_entry_moderate en utf8mb4 COLLATE utf8mb4_unicode_ci en cours. Erreur sur la requête : ALTER TABLE grr_entry_moderate CONVERT TO CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci (1292 : Incorrect datetime value: '0000-00-00 00:00:00' for column 'TIMESTAMP' at row 636)
Erreur sur la requête : ALTER TABLE grr_entry_moderate CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci (1292 : Incorrect datetime value: '0000-00-00 00:00:00' for column 'TIMESTAMP' at row 636)**
Passage de grr_j_mailuser_room en utf8mb4 COLLATE utf8mb4_unicode_ci en cours. Ok !
Passage de grr_j_site_area en utf8mb4 COLLATE utf8mb4_unicode_ci en cours. Ok !
Passage de grr_j_type_area en utf8mb4 COLLATE utf8mb4_unicode_ci en cours. Ok !
Passage de grr_j_user_area en utf8mb4 COLLATE utf8mb4_unicode_ci en cours. Ok !
Passage de grr_j_user_room en utf8mb4 COLLATE utf8mb4_unicode_ci en cours. Ok !
Passage de grr_j_useradmin_area en utf8mb4 COLLATE utf8mb4_unicode_ci en cours. Ok !
Passage de grr_j_useradmin_site en utf8mb4 COLLATE utf8mb4_unicode_ci en cours. Ok !
Passage de grr_j_userbook_room en utf8mb4 COLLATE utf8mb4_unicode_ci en cours. Ok !
Passage de grr_log en utf8mb4 COLLATE utf8mb4_unicode_ci en cours. Ok !
Passage de grr_modulesext en utf8mb4 COLLATE utf8mb4_unicode_ci en cours. Ok !
Passage de grr_overload en utf8mb4 COLLATE utf8mb4_unicode_ci en cours. Ok !
Passage de grr_page en utf8mb4 COLLATE utf8mb4_unicode_ci en cours. Ok !
Passage de grr_participants en utf8mb4 COLLATE utf8mb4_unicode_ci en cours. Ok !
Passage de grr_repeat en utf8mb4 COLLATE utf8mb4_unicode_ci en cours. Ok !
Passage de grr_room en utf8mb4 COLLATE utf8mb4_unicode_ci en cours. Ok !
Passage de grr_setting en utf8mb4 COLLATE utf8mb4_unicode_ci en cours. Ok !
Passage de grr_site en utf8mb4 COLLATE utf8mb4_unicode_ci en cours. Ok !
Passage de grr_type_area en utf8mb4 COLLATE utf8mb4_unicode_ci en cours. Ok !
Passage de grr_utilisateurs en utf8mb4 COLLATE utf8mb4_unicode_ci en cours. Ok !
Mise à jour jusqu'à la version 3.5.1:
**Erreur sur la requête : ALTER TABLE grr_entry_moderate CHANGE login_moderateur login_moderateur varchar(190) CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci NOT NULL default '' ; (1292 : Incorrect datetime value: '0000-00-00 00:00:00' for column 'TIMESTAMP' at row 636)
Erreur sur la requête : ALTER TABLE grr_entry_moderate CHANGE create_by create_by varchar(190) CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci NOT NULL default '' ; (1292 : Incorrect datetime value: '0000-00-00 00:00:00' for column 'TIMESTAMP' at row 636)
Erreur sur la requête : ALTER TABLE grr_entry_moderate CHANGE beneficiaire beneficiaire varchar(190) CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci NOT NULL default '' ; (1292 : Incorrect datetime value: '0000-00-00 00:00:00' for column 'TIMESTAMP' at row 636)**
Résultat de la mise à jour GRR 4 et +
4000 => 0400003
Mise à jour jusqu'à la version 4.0.0 :
Ok !
Mise à jour jusqu'à la version 4.1.0 :
Ok !
Mise à jour jusqu'à la version 4.2.0 :
Ok !
Mise à jour jusqu'à la version 4.3.0 :
**Erreur sur la requête : CREATE TABLE grr_j_userbook_room (login varchar(190) CHARACTER SET utf8mb4 COLLATE utf8mb4_general_ci NOT NULL default '', id_room int(11) NOT NULL default '0', PRIMARY KEY (login,id_room) ) (1050 : Table 'grr_j_userbook_room' already exists)**

**Comment by ynaessens** [comment]
Bonjour,
c'est étrange. Le patch consiste essentiellement à modifier la valeur par défaut du champ timestamp. En mode strict MySQL n'accepte pas la valeur nulle. D'où l'erreur que vous avez et qui risque d'affecter toute la requête, même si elle ne modifie pas timestamp. Si je comprends bien, c'est la ligne 636 de grr_entry_moderate qui pose problème. Pouvez-vous vérifier par phpmyadmin le contenu de cette ligne ?
Pour ce qui est de l'erreur sur la création de grr_j_userbook_room, ce n'est pas bloquant, mais je vais la supprimer pour rassurer les utilisateurs.
Cordialement,
YN

**Reply by Fred12400** [reply]
Voici l'entrée en question : elle est quasiment vide : 

INSERT INTO `grr_entry_moderate` (`id`, `login_moderateur`, `motivation_moderation`, `start_time`, `end_time`, `entry_type`, `repeat_id`, `room_id`, `TIMESTAMP`, `create_by`, `beneficiaire_ext`, `beneficiaire`, `name`, `type`, `description`, `statut_entry`, `option_reservation`, `overload_desc`, `moderate`) VALUES
(8897, 'DIVES', '', 0, 0, 0, 0, 0, '2018-11-15 09:24:16', '', '', '', '', '', '', '', 0, '', 0);

Puis-je la supprimer sans compromettre les autres tables ? Sinon je rentre des informations factices et je relance la MAJ.

**Reply by Fred12400** [reply]
J'ai corrigé l'entrée problématique avec des données fictives, ainsi que les 4 entrées timestamp à 0 et cette fois-ci c'est passé.

Toujours l'erreur create table grr_j_userbook_room mais c'est normal.

Merci de votre aide.

Cdt.
